XamlAnimatedGif/ImageAnimator.cs

```diff
@@ -3,6 +3,7 @@
 using System.Threading.Tasks;
 using System.Windows.Controls;
 using System.Windows.Media.Animation;
+using System.Windows.Shell;
```

Comment:
What is this using? I suspect it was added by mistake, because I see nothing in your changes that would require it.

Anyway, if you get rid of the per-control temp path as per my previous comment, no changes to this class should be necessary.